trace_context/HTTP_HEADER_FORMAT.md

```diff
@@ -147,7 +150,22 @@ base16(<SpanId>) = 00f067aa0ba902b7
 base16(<TraceOptions>) = 00  // not-sampled
 ```
 
-# TraceState field
+## Versioning
+
+Implementation is opinionated about future version of specification. We hope to make format of `traceparent` additive.
+
+Implementation should follow a following rules when parsing headers with unexpected format:
+
+1. Pass thru services should not analyze version. Pass thru service needs to expect that headers may have bigger limits in future.
+2. When version prefix cannot be parsed (it's not 2 hex characters followed by dash), implementation should restart trace.
+3. If higher version is detected - implementation should try to parse.
+  a. If size of header shorter - implementation should not parse it.
+  b. Try parse `trace-id`: from dash 32 characters. Make sure followed by dash.
+  c. Try parse `span-id`: from the second dash 16 characters. Make sure followed by dash.
+  d. Try parse sampling bit of `flags`:  2 characters from third dash. Following with either end of string or a dash.
+  If all three values were parsed successfully - implementation should use them. After parsing - implementation MUST NOT assume anything about other parts of `traceparent` and MUST send version `00` to downstream services.
```

Comment:
why 00? Maybe "send the version that application understands"? What about the potential future fields after the flags - keep them or drop them? (if sending earlier version then have to drop them)

Reply:
I put a note that it is for the current specification